Source/Libraries/GSF.Web/Model/ModelController.cs

```diff
@@ -43,7 +44,16 @@ namespace GSF.Web.Model
     /// Base Class for A Webcontroller that provides common Model endpoints such as Edit, Delete, Search.
     /// </summary>
     /// <typeparam name="T">The corresponding Model.</typeparam>
-    public class ModelController<T> : ApiController where T : class, new()
+    public class ModelController<T> : ModelController<T, T> where T : class, new() { }
+
+    /// <summary>
+    /// Base Class for A Webcontroller that provides common Model endpoints such as Edit, Delete, Search.
+    /// </summary>
+    /// <typeparam name="T">The corresponding Model for Search/Fetch Results.</typeparam>
+    /// <typeparam name="U">The corresponding Model for database editing.</typeparam>
+    public class ModelController<T,U> : ApiController 
+        where T : class, new() 
+        where U : class, new()
```

Comment:
would it be worth further restricting U to be `U: T, new()`?
The use case would be to only allow this for Views that include the entire Table.
That would also allow you to use a single `PrimaryKeyField`

Reply:
As per latest commit, T is now set to be a child of U.